airflow/ui/src/components/SearchDags.tsx

```diff
@@ -0,0 +1,99 @@
+/*!
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { Field } from "@chakra-ui/react";
+import { useQueryClient } from "@tanstack/react-query";
+import { AsyncSelect } from "chakra-react-select";
+import type { OptionsOrGroups, GroupBase, OnChangeValue } from "chakra-react-select";
+import debounce from "debounce-promise";
+import React from "react";
+import { useNavigate } from "react-router-dom";
+
+import { UseDagServiceGetDagsKeyFn } from "openapi/queries";
+import { DagService } from "openapi/requests/services.gen";
+import type { DAGCollectionResponse, DAGResponse } from "openapi/requests/types.gen";
+
+import { DropdownIndicator } from "./SearchDagsDropdownIndicator";
+
+export type Option = {
+  label: string;
+  value: string;
+};
+
+export const SearchDags = ({
+  setIsOpen,
+}: {
+  readonly setIsOpen: React.Dispatch<React.SetStateAction<boolean>>;
+}) => {
+  const queryClient = useQueryClient();
+  const navigate = useNavigate();
+  const SEARCH_LIMIT = 10;
+
+  const onSelect = (newValue: OnChangeValue<Option, false>) => {
+    const selected = newValue as Option;
+
+    if (newValue) {
+      setIsOpen(false);
+      navigate(`/dags/${selected.value}`);
+    }
+  };
+
+  const searchDag = (
+    inputValue: string,
+    callback: (options: OptionsOrGroups<Option, GroupBase<Option>>) => void,
+  ): Promise<OptionsOrGroups<Option, GroupBase<Option>>> =>
+    queryClient.fetchQuery({
+      queryFn: () =>
+        DagService.getDags({
+          dagDisplayNamePattern: inputValue,
+          limit: SEARCH_LIMIT,
+        }).then((data: DAGCollectionResponse) => {
+          const options = data.dags.map(
+            (dag: DAGResponse) =>
+              ({
+                label: dag.dag_display_name || dag.dag_id,
+                value: dag.dag_id,
+              }) as Option,
```

Comment:
```suggestion
              }),
```

We shouldn't need to use `as`

Reply:
Thanks, applied the suggestion locally.